lib/usdUfe/ufe/UsdClipboard.cpp

```diff
@@ -0,0 +1,129 @@
+{
+    // Note: if a clipboard file already exists, it automatically gets overridden, so there is no
+    // need to clear it.
+    // Note: export the root layer directly as the stage export will flatten which removes
+    //       variant sets, payloads, etc.
+    PXR_NS::SdfFileFormat::FileFormatArguments args;
+    args[PXR_NS::UsdUsdFileFormatTokens->FormatArg] = _clipboardFileFormat;
+    if (!clipboardData->GetRootLayer()->Export(_clipboardFilePath, "UsdUfe clipboard", args)) {
+        const std::string error
+            = "Failed to export Clipboard stage with destination: " + _clipboardFilePath + ".";
+        throw std::runtime_error(error);
+    }
+
+    // Unload the stage, otherwise when we try to set and get the next clipboard data we end up with
+    // the old stage.
+    clipboardData->Unload();
+}
+
+PXR_NS::UsdStageWeakPtr UsdClipboard::getClipboardData()
+{
+    // Check if the layer exists
+    auto layer = PXR_NS::SdfLayer::FindOrOpen(_clipboardFilePath);
+    if (!layer)
+        return {};
+
+    PXR_NS::UsdStageRefPtr clipboardStage = PXR_NS::UsdStage::Open(_clipboardFilePath);
+
+    cleanClipboardStageCache();
+
+    // Add the clipboard USD stage to UsdUtilsStageCache, otherwise it is destroyed once out of
+    // scope.
+    _clipboardStageCacheId = PXR_NS::UsdUtilsStageCache::Get().Insert(clipboardStage);
+
+    // Force the new stage to reload, so we don't end up with the old stage.
+    clipboardStage->Reload();
+
+    return clipboardStage;
+}
+
+void UsdClipboard::setClipboardPath(const std::string& clipboardPath)
+{
+    auto tmpPath = std::filesystem::path(clipboardPath);
+    tmpPath.append(clipboardFileName);
+    setClipboardFilePath(tmpPath.string());
+}
+
+void UsdClipboard::setClipboardFilePath(const std::string& clipboardFilePath)
+{
+    _clipboardFilePath = clipboardFilePath;
+}
+
+void UsdClipboard::setClipboardFileFormat(const std::string& formatTag)
+{
+    _clipboardFileFormat = formatTag;
+}
+
+void UsdClipboard::cleanClipboard()
+{
+    cleanClipboardStageCache(true);
+    removeClipboardFile();
+}
+
+void UsdClipboard::cleanClipboardStageCache(bool cleanFile)
+{
+    // Erase the clipboard stage from the cache.
+    if (_clipboardStageCacheId.IsValid()) {
+        auto clipboardStageRef = PXR_NS::UsdUtilsStageCache::Get().Find(_clipboardStageCacheId);
+        PXR_NS::UsdUtilsStageCache::Get().Erase(clipboardStageRef);
+
+        if (cleanFile && clipboardStageRef && clipboardStageRef->GetRootLayer()) {
+            // Clean the stage and clipboard file.
+            // Note: Instead of clearing the file, it might be better to use filenames
+            // from the stage hash and retain that in a variable or stack.
+            clipboardStageRef->GetRootLayer()->Clear();
+            PXR_NS::SdfFileFormat::FileFormatArguments args;
+            args["format"] = _clipboardFileFormat;
+            clipboardStageRef->GetRootLayer()->Export(_clipboardFilePath, "UsdUfe Clipboard", args);
```

Comment:
So the goal here is to generate a new file that contains nothing? What is the purpose vs just deleting the file?

Reply:
Yes good idea Alice. I've made those changes.